src/main/java/thePackmaster/patches/gemspack/AddEnhanceButtonPatch.java

```diff
@@ -0,0 +1,46 @@
+package thePackmaster.patches.gemspack;
+
+import com.evacipated.cardcrawl.modthespire.lib.*;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.rooms.CampfireUI;
+import com.megacrit.cardcrawl.ui.campfire.AbstractCampfireOption;
+import javassist.CtBehavior;
+import thePackmaster.ThePackmaster;
+import thePackmaster.packs.GemsPack;
+import thePackmaster.ui.EnhanceBonfireOption;
+import thePackmaster.vfx.gemspack.SocketGemEffect;
+
+import java.util.ArrayList;
+
+public class AddEnhanceButtonPatch {
+    @SpirePatch(clz = CampfireUI.class, method = "initializeButtons")
+    public static class AddKeys {
+        @SpireInsertPatch(locator = Locator.class)
+        public static void patch(CampfireUI __instance, ArrayList<AbstractCampfireOption> ___buttons) {
+            Boolean active = true;
+            if (SocketGemEffect.getModifiableCards().size() == 0) {
+                active = false;
+            }
+            if (SocketGemEffect.getGems().size() == 0) {
+                active = false;
+            }
+            if (AbstractDungeon.player.gold < GemsPack.goldCostToSocket) {
+                active = false;
+            }
+            if (AbstractDungeon.player instanceof ThePackmaster || active) {
```

Comment:
So this always shows on the Packmaster regardless? I don't like that to be honest.

If you really want to, I'd be fine if it showed if the gems pack were in the current packs.

Reply:
Yeah, fair, it isn't necessary at all to check the character here.